src/DiveWish.spec.js

```diff
@@ -14,10 +15,44 @@ describe('ListItem', () => {
     const notes = screen.getByText(
       'I want to go there for my next diving holiday'
     );
+    const buttonBookmark = screen.getByRole('button', { name: /bookmark/i });
     const buttonDelete = screen.getByRole('button', { name: /delete/i });
 
     expect(destination).toBeInTheDocument();
     expect(notes).toBeInTheDocument();
     expect(buttonDelete).toBeInTheDocument();
+    expect(buttonBookmark).toBeInTheDocument();
   });
+
+  it('clicking the bookmark button calls the function to toggle the bookmark', () => {
+    const toggleBookmark = jest.fn();
+    render(
+      <DiveWish
+        destination="Maldives"
+        notes="I want to go there for my next diving holiday"
+        onToggleBookmark={toggleBookmark}
+      />
+    );
+
+    const buttonBookmark = screen.getByRole('button', { name: /bookmark/i });
+    userEvent.click(buttonBookmark);
+
+    expect(toggleBookmark).toHaveBeenCalled();
+  });
+
+  // it('clicking the delete button calls the function to show the delete confirmation dialog', () => {
+  //   const showDialog = jest.fn();
+  //   render(
+  //     <DiveWish
+  //       destination="Maldives"
+  //       notes="I want to go there for my next diving holiday"
+  //       onDeleteDiveWish={showDialog}
+  //     />
+  //   );
+
+  //   const buttonDelete = screen.getByRole('button', { name: /delete/i });
+  //   userEvent.click(buttonDelete);
+
+  //   expect(showDialog).toHaveBeenCalled();
```

Comment:
You are missing one step here. If you click on the `delete` button, then the modal opens and you have to confirm the deletion and only then the `onDeleteDiveWish` function will be called. So after you click on the `buttonDelete` you should query for your confirmation button and check that it calls your callback function. I guess you'd also need another test, that tests if you click on the cancel button, the modal will close and not be found. If you need help, please feel free to ask me :)